addons/addon-base-raas/packages/base-raas-services/lib/plugins/env-provisioning-plugin.js

```diff
@@ -220,6 +220,29 @@ async function updateEnvOnProvisioningSuccess({
       const environmentDnsService = await container.find('environmentDnsService');
       await environmentDnsService.createRecord('rstudio', envId, dnsName);
     }
+    else if (connectionTypeValue.toLowerCase() === 'rstudiov2') {
+      const albService = await container.find('albService');
+      const deploymentItem = await albService.getAlbDetails(requestContext, resolvedVars.projectId);
+      const albExists = await albService.checkAlbExists(requestContext, resolvedVars.projectId);
+      if (!albExists) {
+        throw new Error(`Error provisioning environment. Reason: No ALB found for this AWS account`);
+      }
+      const dnsName = JSON.parse(deploymentItem.value).albDnsName;
+      const targetGroupArn = _.find(outputs, o => o.OutputKey === 'TargetGroupARN').OutputValue;
+      // Create DNS record for RStudio workspaces
+      const environmentDnsService = await container.find('environmentDnsService');
+      await environmentDnsService.createRecord('rstudio', envId, dnsName);
+      //Create a listener rule
+      const ruleARN = await albService.createListenerRule('rstudio', requestContext, resolvedVars, targetGroupArn);
+      // Save the Rule ARN as an output in DB
+      const ruleRecord = {
+        Description: "ARN of the listener rule created by code",
+        OutputKey: "ListenerRuleARN",
+        OutputValue: ruleARN
+      };
+      outputs.push(ruleRecord);
+      await albService.increaseAlbDependentWorkspaceCount(requestContext, resolvedVars);
```

Comment:
Seems like we need to put this code under a distributed lock so that we don't potentially miss an update. Take a look at [lock-service](https://github.com/awslabs/service-workbench-on-aws/blob/mainline/addons/addon-base/packages/services/lib/lock/lock-service.js) and [example code](https://github.com/awslabs/service-workbench-on-aws/blob/mainline/addons/addon-base-raas/packages/base-raas-services/lib/aws-accounts/aws-accounts-service.js#L91)

Reply:
The concurrent request issues will be resolved once the lock service is implemented. You can expect that in the next PR